scenarios/csharp/dotnet/captioning/Program.cs

```diff
@@ -0,0 +1,417 @@
+/*
+Dependencies
+*/
+
+using System; // File
+using System.Linq; // Enumerable.Contains
+using System.Text; // StringBuilder
+using System.Threading.Tasks; // Task
+
+/* Notes:
+- Install the Speech NuGet package:
+dotnet add package Microsoft.CognitiveServices.Speech
+- The Speech SDK on Windows requires the Microsoft Visual C++ Redistributable for Visual Studio 2019 on the system. See:
+https://docs.microsoft.com/azure/cognitive-services/speech-service/speech-sdk?tabs=windows%2Cubuntu%2Cios-xcode%2Cmac-xcode%2Candroid-studio
+*/
+using Microsoft.CognitiveServices.Speech;
+using Microsoft.CognitiveServices.Speech.Audio;
+
+namespace Caption
+{
+/*
+Types
+*/
+
+    public struct UserConfig
+    {
+        readonly public bool profanityFilterRemoveEnabled = false;
+        readonly public bool profanityFilterMaskEnabled = false;
+        readonly public string[]? languageIDLanguages;
+        readonly public string? inputFile;
+        readonly public string? outputFile;
+        readonly public string? phraseList;
+        readonly public bool suppressOutputEnabled = false;
+        readonly public bool partialResultsEnabled = false;
+        readonly public string? stablePartialResultThreshold;
+        readonly public bool srtEnabled = false;
+        readonly public bool trueTextEnabled = false;
+        readonly public string subscriptionKey;
+        readonly public string region;
+        
+        public UserConfig (
+            bool profanityFilterRemoveEnabled,
+            bool profanityFilterMaskEnabled,
+            string[]? languageIDLanguages,
+            string? inputFile,
+            string? outputFile,
+            string? phraseList,
+            bool suppressOutputEnabled,
+            bool partialResultsEnabled,
+            string? stablePartialResultThreshold,
+            bool srtEnabled,
+            bool trueTextEnabled,
+            string subscriptionKey,
+            string region
+            )
+        {
+            this.profanityFilterRemoveEnabled = profanityFilterRemoveEnabled;
+            this.profanityFilterMaskEnabled = profanityFilterMaskEnabled;
+            this.languageIDLanguages = languageIDLanguages;
+            this.inputFile = inputFile;
+            this.outputFile = outputFile;
+            this.phraseList = phraseList;
+            this.suppressOutputEnabled = suppressOutputEnabled;
+            this.partialResultsEnabled = partialResultsEnabled;
+            this.stablePartialResultThreshold = stablePartialResultThreshold;
+            this.srtEnabled = srtEnabled;
+            this.trueTextEnabled = trueTextEnabled;
+            this.subscriptionKey = subscriptionKey;
+            this.region = region;
+        }
+    }
+
+    public class TooFewArguments : Exception
+    {
+        public TooFewArguments() : base() { }
+    }
+
+    class Program
+    {
+
+/*
+Helper functions
+*/
+
+        static string? GetCmdOption(string[] args, string option)
```

Comment:
Instance feedback: it's good practice to apply an access modifier (public/private) all the time
Broader/more general: it may be helpful to compile this with static analysis (default rules) turned on to see what it yields -- it'll include things like this instance feedback along with other things like public/private ordering, property/field/method ordering, etc.